sharktank/tests/models/llama/sharded_llama_test.py

```diff
@@ -237,5 +348,86 @@ def _(model, *args, **kwargs) -> torch.Tensor:
             def _(model, *args, **kwargs) -> torch.Tensor:
                 return model.decode(*args, **kwargs)
 
+            # Compile the IREE module.
             output = export(sharded_fxb)
             output.save_mlir(f"{temp_dir}/program.mlir")
+            output.session.set_flags(
+                *[
+                    f"--iree-hal-target-device=llvm-cpu[{i}]"
+                    for i in range(self.sharded_config.tensor_parallelism_size)
+                ]
+            )
+            iree_module_path = f"{temp_dir}/program.vmfb"
+            output.compile(
+                save_to=iree_module_path,
+                target_backends=None,
+            )
+
+            iree_devices = get_iree_devices(self.sharded_config.tensor_parallelism_size)
+            iree_module, vm_context, vm_instance = load_iree_module(
+                module_path=iree_module_path,
+                devices=iree_devices,
+                parameters_path=sharded_parameters_path,
+            )
+
+            # Check IREE's prefill step is close to torch.
+            prefill_iree_args = prepare_iree_module_function_args(
+                args=deepcopy(sharded_prefill_args).values(), devices=iree_devices
+            )
+            prefill_iree_result = run_iree_module_function(
+                args=prefill_iree_args,
+                function_name="prefill",
+                module=iree_module,
+                vm_context=vm_context,
+            )
+            prefill_iree_result = iree_to_torch(*prefill_iree_result)
+            assert len(prefill_iree_result) == 1
+            expected_prefill_result = sharded_model.prefill(**sharded_prefill_args)
+            # TODO: Although, not entirely wrong, investigate why this accuracy is that
+            # low for fp32.
+            torch.testing.assert_close(
+                prefill_iree_result[0], expected_prefill_result, atol=0.0011, rtol=0.013
```

Comment:
This is probably fine but I think I would rather see stricter tolerances that we should be able to meet, and xfail until it works as expected.

Reply:
Make sense. I changed to default tolerance.